klog.go

```diff
@@ -1225,24 +1226,36 @@ func (l *loggingT) lockAndFlushAll() {
 	l.mu.Lock()
 	l.flushAll()
 	l.mu.Unlock()
+	// Some environments are slow when syncing and holding the lock might cause contention.
+	l.syncAll()
 }
 
-// flushAll flushes all the logs and attempts to "sync" their data to disk.
+// flushAll flushes all the logs
 // l.mu is held.
 func (l *loggingT) flushAll() {
 	// Flush from fatal down, in case there's trouble flushing.
 	for s := severity.FatalLog; s >= severity.InfoLog; s-- {
 		file := l.file[s]
 		if file != nil {
 			_ = file.Flush() // ignore error
-			_ = file.Sync()  // ignore error
 		}
 	}
 	if logging.loggerOptions.flush != nil {
 		logging.loggerOptions.flush()
 	}
 }
 
+// syncAll attempts to "sync" their data to disk.
+func (l *loggingT) syncAll() {
+	// Flush from fatal down, in case there's trouble flushing.
+	for s := severity.FatalLog; s >= severity.InfoLog; s-- {
+		file := l.file[s]
```

Comment:
I think accessing `l.file[s]` must be protected by the mutex. So lock the mutex, collect all non-nil files in a slice, then unlock and sync them.

Reply:
You could let `flushAll` return a slice of open files. Then pass that into `syncAll` and use it there without locking.